packages/beacon-node/src/chain/blocks/index.ts

```diff
@@ -58,92 +50,41 @@ export class BlockProcessor {
  *
  * All other effects are provided by downstream event handlers
  */
-export async function processBlock(
+export async function processBlocks(
   chain: ProcessBlockModules,
-  partiallyVerifiedBlock: PartiallyVerifiedBlock,
-  opts: BlockProcessOpts
+  blocks: allForks.SignedBeaconBlock[],
+  opts: BlockProcessOpts & ImportBlockOpts
 ): Promise<void> {
+  if (blocks.length === 0) {
+    return; // TODO: or throw?
+  } else if (blocks.length > 1) {
+    assertLinearChainSegment(chain.config, blocks);
+  }
+
   try {
-    const fullyVerifiedBlock = await verifyBlock(chain, partiallyVerifiedBlock, opts);
-    await importBlock(chain, fullyVerifiedBlock);
-  } catch (e) {
-    // above functions should only throw BlockError
-    const err = getBlockError(e, partiallyVerifiedBlock.block);
+    for (const block of blocks) {
+      const fullyVerifiedBlock = await verifyBlock(chain, block, opts);
 
-    if (
-      partiallyVerifiedBlock.ignoreIfKnown &&
-      (err.type.code === BlockErrorCode.ALREADY_KNOWN || err.type.code === BlockErrorCode.GENESIS_BLOCK)
-    ) {
-      // Flag ignoreIfKnown causes BlockErrorCodes ALREADY_KNOWN, GENESIS_BLOCK to resolve.
-      // Return before emitting to not cause loud logging.
-      return;
+      // No need to sleep(0) here since `importBlock` includes a disk write
+      // TODO: Consider batching importBlock too if it takes significant time
+      await importBlock(chain, fullyVerifiedBlock, opts);
     }
+  } catch (e) {
+    // above functions should only throw BlockError
+    const err = getBlockError(e, blocks[0]);
```

Comment:
Shouldn't the block error be for the block that errored?

Reply:
After completing this work everything is processed in batch so you don't know which block caused the error. Each function should take care of adding the right block to BlockError which is already happening, but here, at this line you have no clue which block caused this. blocks[0] is a placeholder, it shouldn't really happen. In future PRs we can review what to do in this case